lib/types.nix

```
@@ -119,8 +119,22 @@ rec {
       #  - "composite": a phrase with an "of" connective
       # See the `optionDescriptionPhrase` function.
     , descriptionClass ? null
-    , # Function applied to each definition that should return true if
-      # its type-correct, false otherwise.
+    , # DO NOT USE WITHOUT KNOWING WHAT YOU ARE DOING
+      # Function applied to each definition that must return false when a definition
+      # does not match the type. It should not check more than the root of the value,
+      # because evaluating contained reduces laziness, leading to unnecessary
+      # infinite recursions in the module system.
+      # Further checks of contained values should be performed by throwing in
```

```suggestion
      # Further checks of nested values should be performed by throwing in
```

... to keep consistency. I'm not religious about if you say nested or contained, just choose one - with with technical terms we should try to be unambiguous whether we mean the same thing by using the same words.